crates/kubelet/src/state/common/allocated.rs

```diff
@@ -0,0 +1,126 @@
+//! Resources can be successfully allocated to the Pod.
+use crate::device_plugin_manager::resources;
+use crate::pod::state::prelude::*;
+use crate::provider::DevicePluginSupport;
+use crate::volume::{HostPathVolume, VolumeRef};
+use k8s_openapi::api::core::v1::HostPathVolumeSource;
+use k8s_openapi::api::core::v1::Volume as KubeVolume;
+use k8s_openapi::apimachinery::pkg::api::resource::Quantity;
+use std::collections::HashMap;
+use tracing::{debug, error, info};
+
+use super::error::Error;
+use super::image_pull::ImagePull;
+use super::{GenericPodState, GenericProvider};
+
+/// Resources can be successfully allocated to the Pod
+pub struct Allocated<P: GenericProvider> {
+    phantom: std::marker::PhantomData<P>,
+}
+
+impl<P: GenericProvider> std::fmt::Debug for Allocated<P> {
+    fn fmt(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        "Allocated".fmt(formatter)
+    }
+}
+
+impl<P: GenericProvider> Default for Allocated<P> {
+    fn default() -> Self {
+        Self {
+            phantom: std::marker::PhantomData,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl<P: GenericProvider> State<P::PodState> for Allocated<P> {
+    async fn next(
+        self: Box<Self>,
+        provider_state: SharedState<P::ProviderState>,
+        pod_state: &mut P::PodState,
+        pod: Manifest<Pod>,
+    ) -> Transition<P::PodState> {
+        let pod = pod.latest();
+        debug!(
+            "Preparing to allocate resources for the pod: {}",
+            pod.name()
```

Comment:
Add this as a field on the log

Reply:
Is this (b1f3a819d19) what you were envisioning? Do you have logging/tracing syntax guidelines?

Reply:
Yeah that is exactly what I was thinking of